app/src/main/java/com/example/sharingang/DetailedItemFragment.kt

```diff
@@ -73,6 +79,62 @@ class DetailedItemFragment : Fragment() {
         return binding.root
     }
 
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        super.onCreateOptionsMenu(menu, inflater)
+        inflater.inflate(R.menu.menu_detailed, menu)
+    }
+
+    override fun onPrepareOptionsMenu(menu: Menu) {
+        super.onPrepareOptionsMenu(menu)
+        val sell = menu.findItem(R.id.menuSell)
+        val resell = menu.findItem(R.id.menuResell)
+        if (!args.item.userId.equals(currentUserProvider.getCurrentUserId())) {
+            menu.findItem(R.id.menuEdit).isVisible = false
+            sell.isVisible = false
+            resell.isVisible = false
+        } else {
+            sell.isVisible = !(item?.sold ?: args.item.sold)
+            resell.isVisible = item?.sold ?: args.item.sold
```

Comment:
```suggestion
            resell.isVisible = item?.sold ?: args.item.sold
            sell.isVisible = !resell.isVisible
```

How about this, to avoid repeating the logic?

Reply:
Sure good idea!